erpnext/manufacturing/doctype/bom/bom.py

```diff
@@ -252,10 +260,16 @@ def validate_materials(self):
 				validate_bom_no(m.item_code, m.bom_no)
 			if flt(m.qty) <= 0:
 				frappe.throw(_("Quantity required for Item {0} in row {1}").format(m.item_code, m.idx))
-			check_list.append(cstr(m.item_code))
-		unique_chk_list = set(check_list)
-		if len(unique_chk_list)	!= len(check_list):
-			frappe.throw(_("Same item has been entered multiple times."))
+			check_list.append(m)
+
+		duplicate_items = list(get_duplicates(check_list))
+		if duplicate_items:
+			li = []
+			for i in duplicate_items:
+				li.append("<li>{0} on row {1}".format(i.item_code, i.idx))
```

Comment:
close the li tag `<li>{0} on row {1}</li>`

Also can we keep the message simple? Just join the list of items with <br>?